src/partials/PoWDifficulty.svelte

```diff
@@ -0,0 +1,54 @@
+<script lang="ts">
+  import {session} from "@welshman/app"
+  import {own} from "@welshman/signer"
+  import {createEvent} from "@welshman/util"
+  import {onMount} from "svelte"
+  import {addPoWStamp} from "src/util/pow"
+  import Field from "src/partials/Field.svelte"
+  import Input from "src/partials/Input.svelte"
+  import PowWorker from "src/workers/pow?worker"
+  import {isMobile} from "src/util/html"
+
+  export let value
+
+  // pow generation time for a 16 difficulty in ms
+  let benchmark = 0
+  const benchmarkDifficulty = isMobile ? 14 : 16
+
+  onMount(() => {
+    // try to generate a simple pow to estimate the device capacities
+    const powWorker = new PowWorker()
+    const ownedEvent = own(
+      createEvent(1, {
+        content: "this is a random pow test",
+        tags: [],
+      }),
+      $session.pubkey,
+    )
+    const start = Date.now()
+    addPoWStamp(powWorker, ownedEvent, benchmarkDifficulty).then(_ => {
+      benchmark = Date.now() - start
+    })
+    return () => powWorker.terminate()
+  })
```

Comment:
This should only need to happen once ever, add this to engine/state and use `synced` to keep the benchmark in localstorage. Also make sure to handle the UI when benchmark = 0 (maybe something like "calculating...")